backend/plugins/pluginservice.go

```diff
@@ -104,181 +102,153 @@ func (s *Service) GetHandler(w http.ResponseWriter, r *http.Request) {
 	commons.WriteJsonResponse(w, plugin)
 }
 
+type AddPluginBody struct {
+	URL string `json:"url"`
+}
+
 func (s *Service) AddHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
This function needs to be refactored into smaller pieces.
JSON decoding of the request and response writing should not be mixed with the rest...